File: http/http-server-annotation-processor/src/main/java/ru/tinkoff/kora/http/server/annotation/processor/extension/HttpServerRequestMapperKoraExtensionFactory.java
package ru.tinkoff.kora.http.server.annotation.processor.extension;

import ru.tinkoff.kora.http.server.annotation.processor.HttpServerClassNames;
import ru.tinkoff.kora.kora.app.annotation.processor.extension.ExtensionFactory;
import ru.tinkoff.kora.kora.app.annotation.processor.extension.KoraExtension;

import javax.annotation.processing.ProcessingEnvironment;
import java.util.Optional;

public final class HttpServerRequestMapperKoraExtensionFactory implements ExtensionFactory {
    @Override
    public Optional<KoraExtension> create(ProcessingEnvironment processingEnvironment) {
        var httpServerRequestMapper = processingEnvironment.getElementUtils().getTypeElement(HttpServerClassNames.httpServerRequestMapper.canonicalName());
        if (httpServerRequestMapper == null) {
            return Optional.empty();
        }
        return Optional.of(new HttpServerRequestMapperKoraExtension(processingEnvironment.getElementUtils(), processingEnvironment.getTypeUtils()));
    }
}

File: http/http-server-symbol-processor/src/main/kotlin/ru/tinkoff/kora/http/server/symbol/procesor/extension/HttpServerRequestMapperKoraExtensionFactory.kt
package ru.tinkoff.kora.http.server.symbol.procesor.extension

import com.google.devtools.ksp.getClassDeclarationByName
import com.google.devtools.ksp.processing.CodeGenerator
import com.google.devtools.ksp.processing.KSPLogger
import com.google.devtools.ksp.processing.Resolver
import ru.tinkoff.kora.http.server.symbol.procesor.HttpServerClassNames
import ru.tinkoff.kora.kora.app.ksp.extension.ExtensionFactory
import ru.tinkoff.kora.kora.app.ksp.extension.KoraExtension

class HttpServerRequestMapperKoraExtensionFactory : ExtensionFactory {
    override fun create(resolver: Resolver, kspLogger: KSPLogger, codeGenerator: CodeGenerator): KoraExtension? {
        if (resolver.getClassDeclarationByName(HttpServerClassNames.httpServerRequestMapper.canonicalName) != null) {
            return HttpServerRequestMapperKoraExtension()
        }
        return null
    }
}
